rdf-delta-server-local/src/main/java/org/seaborne/delta/server/local/patchstores/PatchLogIndex.java

```diff
@@ -33,7 +33,7 @@
     public void runWithLock(Runnable action);
 
     /** Run action inside a patch log wide lock; return a result. */
-    public <X> X runWithLockRtn(Supplier<X> action);
+    public <X> X runWithLock(Supplier<X> action);
```

Comment:
I was told that Java couldn't tell the difference between `Runnable` and `Supplier<X>`, so who knows what chaos would be unleashed switching to a `Callable`?

Reply:
I don't think that's true. Because of type erasure Java can't tell the difference between the same raw type with two different type parameters i.e., you can't do this:
```
void f(Optional<Integer> x)
void f(Optional<Long> x)
```
because those methods have the same signature after type erasure.
But using two different raw types is fine.